.circleci/config.yml

```diff
@@ -0,0 +1,136 @@
+version: 2.1
+
+
+commands:
+  npm-install:
+    steps:
+      - run: sudo npm install npm@6 --global
+      - restore_cache:
+          key: js-deps-{{ checksum "package.json" }}
+      - run: npm install --no-save
+      - save_cache:
+          key: js-deps-{{ checksum "package.json" }}
+          paths:
+            - ./node_modules
+
+  pip-install-docs:
+    steps:
+      - run: sudo pip install --upgrade pip
+      - restore_cache:
+          key: py-deps-{{ checksum "docs/requirements.txt" }}
+      - run: "if [ ! -d ./venv ]; then python -m venv ./venv; fi"
+      - run: echo "source $(pwd)/venv/bin/activate" >> $BASH_ENV
+      - run: pip install -r ./docs/requirements.txt
+      - save_cache:
+          key: py-deps-{{ checksum "docs/requirements.txt" }}
+          paths:
+            - ./venv
+
+
+jobs:
+  test-node12:
+    docker:
+      - image: circleci/node:12
+    steps:
+      - checkout
+      - npm-install
+      - run: npm run ci:test
```

Comment:
These are all repeated steps below, I'd suggest using an reusable component:

See https://github.com/apiaryio/api-elements.js/blob/38fdacc56a1a7ef3a8e238e3852ccc987b673996/.circleci/config.yml for working example.

```yaml
aliases:
  - &defaults
    steps:
      - checkout
      - npm-install
      - run: npm run ci:test

jobs:
  test-node12:
    docker: [image: circleci/node:12]
    <<: *defaults

  test-node10:
    docker: [image: circleci/node:10]
    <<: *defaults
```